mlflow/utils/cli_args.py

```diff
@@ -45,12 +48,85 @@
 NO_CONDA = click.option(
     "--no-conda",
     is_flag=True,
-    help="If specified, will assume that MLmodel/MLproject is running within "
+    help="[Deprecated] If specified, will assume that MLmodel/MLproject is running within "
     "a Conda environment with the necessary dependencies for "
     "the current project instead of attempting to create a new "
     "conda environment.",
 )
 
+
+def _env_manager_callback(_, __, value):
+    """
+    Validates the value of `--env-manager` and converts it to the corresponding member of
+    `EnvManager`.
+    """
+    if value is None:
+        return value
+
+    allowed_values = [e.value for e in EnvManager]
+    if value not in allowed_values:
+        raise click.BadParameter(f"Expected one of {allowed_values} but got '{value}'")
+
+    return EnvManager[value.upper()]
+
+
+ENV_MANAGER = click.option(
+    "--env-manager",
+    default=None,
+    required=False,
+    type=click.UNPROCESSED,
+    callback=_env_manager_callback,
+    help="If specified, create an environment for MLmodel/MLproject using the specified "
+    "environment management tool. Valid values are ['local', 'conda', 'virtualenv']. "
+    "If unspecified, default to 'conda'.",
+)
+
+
+def _get_env_manager(no_conda, env_manager):
```

Comment:
I suggest adding a test for `_get_env_manager`. The logic of the function is a little complex.

Reply:
Added tests!